### DISCUSSION TITLE
Maui xaml是不是有控件数量上限？

### DISCUSSION THREAD

**Discussion by sfmmwl** [BYJRK/DotNet-Discussions#42] [Q&A] [answered]
你好，我是c#编程爱好者，目前正在尝试用微软新出的maui与你视频中的mvvm toolkit写for win和for android的小程序。
昨天碰到一个调试运行错误，错误描述是

{"A cycle occurred while laying out the GUI."}
Layout cycle detected.  Layout could not complete.

在百度和bing上搜索相关信息，好像是说控件添加太多导致程序崩溃，估计与我创建的控件模板和xaml中静态添加该模板（大约一个page有100多处）有关。
但没有找到具体解决方法。目前只能再尝试替换为用collectionview和toolkit message动态添加该控件模板，也不知道是不是能解决该问题。

不知道朋友你在实际开发中有没有碰到过该错误，能不能分享解决方法？

**Comment by Dynesshely** [comment, marked as answer]
从字面来看, 您遇到的报错意思是: 检测到了循环布局, 无法完成布局
请检查您的控件模板是否有非法嵌套导致了循环布局的发生

**Reply by sfmmwl** [reply]
谢谢回复。
我也bing过相关问题，基本上说是可能控件数量太多，导致框架渲染用时过长，框架就认为是死循环了。
没办法，现在我简化了相关模板，放弃了一些功能，错误消失。

**Reply by Dynesshely** [reply]
是的, 逻辑层面没法直接判断是否存在嵌套布局, 通过计算布局的时间来判断是否包含嵌套是一个可选的方式
